foxglove_bridge_base/include/foxglove_bridge/websocket_server.hpp

```diff
@@ -602,4 +639,51 @@ inline std::optional<asio::ip::tcp::endpoint> Server<ServerConfiguration>::local
   return endpoint;
 }
 
+template <>
+bool Server<WebSocketNoTls>::USES_TLS = false;
+
+template <>
+bool Server<WebSocketTls>::USES_TLS = true;
+
+template <>
+inline void Server<WebSocketNoTls>::setupTlsHandler() {
+  _server.get_alog().write(APP, "Server running without TLS");
+}
+
+template <>
+inline void Server<WebSocketTls>::setupTlsHandler() {
+  _server.set_tls_init_handler([this](ConnHandle hdl) {
+    (void)hdl;
+
+    namespace asio = websocketpp::lib::asio;
+    auto ctx = websocketpp::lib::make_shared<asio::ssl::context>(asio::ssl::context::sslv23);
+
+    try {
+      ctx->set_options(asio::ssl::context::default_workarounds | asio::ssl::context::no_tlsv1 |
+                       asio::ssl::context::no_sslv2 | asio::ssl::context::no_sslv3);
+      ctx->use_certificate_chain_file(_certfile);
+      ctx->use_private_key_file(_keyfile, asio::ssl::context::pem);
+
+      constexpr char ciphers[] =
+        "ECDHE-RSA-AES128-GCM-SHA256:ECDHE-ECDSA-AES128-GCM-SHA256:ECDHE-RSA-AES256-GCM-SHA384:"
+        "ECDHE-ECDSA-AES256-GCM-SHA384:DHE-RSA-AES128-GCM-SHA256:DHE-DSS-AES128-GCM-SHA256:kEDH+"
+        "AESGCM:ECDHE-RSA-AES128-SHA256:ECDHE-ECDSA-AES128-SHA256:ECDHE-RSA-AES128-SHA:ECDHE-ECDSA-"
+        "AES128-SHA:ECDHE-RSA-AES256-SHA384:ECDHE-ECDSA-AES256-SHA384:ECDHE-RSA-AES256-SHA:ECDHE-"
+        "ECDSA-AES256-SHA:DHE-RSA-AES128-SHA256:DHE-RSA-AES128-SHA:DHE-DSS-AES128-SHA256:DHE-RSA-"
+        "AES256-SHA256:DHE-DSS-AES256-SHA:DHE-RSA-AES256-SHA:AES128-GCM-SHA256:AES256-GCM-SHA384:"
+        "AES128-SHA256:AES256-SHA256:AES128-SHA:AES256-SHA:AES:CAMELLIA:DES-CBC3-SHA:!aNULL:!eNULL:"
+        "!EXPORT:!DES:!RC4:!MD5:!PSK:!aECDH:!EDH-DSS-DES-CBC3-SHA:!EDH-RSA-DES-CBC3-SHA:!KRB5-DES-"
+        "CBC3-SHA";
```

Comment:
Can we provide a link or reference to where this cipher list came from?

Reply:
It stems from the websocketpp example echo tls server:
https://github.com/zaphoyd/websocketpp/blob/1b11fd301531e6df35a6107c1e8665b1e77a2d8e/examples/echo_server_tls/echo_server_tls.cpp#L121

But I actually should have used the more restrictive version 2 lines above since we also disable `tlsv1`. I will update the cipher and also add a link to the example server code.